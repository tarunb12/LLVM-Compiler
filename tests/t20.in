// Shift multiplication

int multiply(int p, int n) {
    int result = 0
    int count = 0
    int q = n
    while (q != 0) {
        if (q % 2 == 1) result += (p << count)
        count++
        q /= 2
    }
    return result
}

int main() {
    int p = 47
    int q = 38
    printf("%d * %d = %d\n", p, q, multiply(p, q))
    p = 32
    q = 87
    printf("%d * %d = %d\n", p, q, multiply(p, q))
    p = 2983
    q = 2012
    printf("%d * %d = %d\n", p, q, multiply(p, q))
}