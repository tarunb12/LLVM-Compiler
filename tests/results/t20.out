Error: File "tests/t20.in": No entry point found in the file. Define the entry point with the function main defined as follows:

int main() {
	...
	return 0
}
.